rct229/rules/section6/section6rule9.py

```diff
@@ -0,0 +1,135 @@
+        )
+
+    def create_data(self, context, data=None):
+        rmd_p = context.proposed
+        return {"is_leap_year": rmd_p["calendar"]["is_leap_year"]}
+
+    class RuleSetModelInstanceRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section6Rule9.RuleSetModelInstanceRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, True),
+                each_rule=Section6Rule9.RuleSetModelInstanceRule.BuildingRule(),
+                index_rmr="proposed",
+                list_path="buildings[*]",
+                required_fields={"$": ["schedules"]},
+            )
+
+        def create_data(self, context, data=None):
+            rmd_p = context.proposed
+            return {
+                **data,
+                "schedules_p": rmd_p["schedules"],
+            }
+
+        class BuildingRule(RuleDefinitionListIndexedBase):
+            def __init__(self):
+                super(Section6Rule9.RuleSetModelInstanceRule.BuildingRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, True),
+                    each_rule=Section6Rule9.RuleSetModelInstanceRule.BuildingRule.ZoneRule(),
+                    index_rmr="proposed",
+                    list_path="$..zones[*]",
+                )
+
+            def is_applicable(self, context, data=None):
+                building_p = context.proposed
+                return pint_sum(find_all("spaces[*].floor_area", building_p), ZERO.AREA) >= FLOOR_AREA_LIMIT
+
+            def create_data(self, context, data=None):
+                building_p = context.proposed
+                schedules_p = data["schedules_p"]
+                return {
+                    **data,
+                    "building_open_schedule_p":  getattr_(
+                        find_one_with_field_value("$", "id", building_p["building_open_schedule"], schedules_p),
+                        "schedule", "hourly_values")
+                }
+
+            class ZoneRule(RuleDefinitionListIndexedBase):
+                def __init__(self):
+                    super(Section6Rule9.RuleSetModelInstanceRule.BuildingRule.ZoneRule, self).__init__(
+                        rmrs_used=UserBaselineProposedVals(False, True, True),
+                        each_rule=Section6Rule9.RuleSetModelInstanceRule.BuildingRule.ZoneRule.SpaceRule(),
+                        index_rmr="proposed",
+                        list_path="spaces[*]",
+                    )
+
+                def create_data(self, context, data=None):
+                    zone_p = context.proposed
+                    return {
+                        **data,
+                        "avg_space_height": zone_p.get("volume", 0.0) / pint_sum(
+                            find_all("spaces[*].floor_area", zone_p), ZERO.AREA)
+                    }
+
+                def list_filter(self, context_item, data=None):
```

Comment:
This doesn't seem to be in the RDS. How would I know to make this restriction?

Since we are indexing on proposed, should the check be for `space_p`?
Or do you want `space_b.get("interior_lighting") and space_p.get("interior_lighting")?

The second argument of `get` defaults to `None`, so the `None` is optional here.

Reply:
My 2 cents was spaces with no interior lighting should be skipped for processing, but I just realize the normalized schedule would take care of this scenario. I will remove this filter.